### PR TITLE
fix broken go get on the kind stage

### PR DESCRIPTION
**Pull request by yastij**
Signed-off-by: Yassine TIJANI <ytijani@vmware.com>

**What this PR does / why we need it**: fixes broken go get on the kind stage, needed to rebuild the CI image

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:

/assign @akutz @frapposelli 

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: hack/images/ci/Dockerfile [hunk 1]
```diff
@@ -52,7 +52,7 @@ RUN /google-cloud-sdk/bin/gcloud components update
 ##                               KIND STAGE                                   ##
 ################################################################################
 FROM ${GOLANG_IMAGE} as kind
-RUN go get -u sigs.k8s.io/kind
+RUN GO111MODULE="on" go get -u sigs.k8s.io/kind@v0.3.0
```
Comment (akutz) [inline hack/images/ci/Dockerfile]: /hold

Hi @yastij,

Won't this update the project's `go.mod` and `go.sum` that are then persisted into the final container?

Comment (akutz) [reply hack/images/ci/Dockerfile]: Maybe update the `RUN` line to `cd / && ...` to ensure it doesn't execute in the project directory.

Comment (yastij) [reply hack/images/ci/Dockerfile]: @akutz - oh right, good catch !

**Pull request merged**